### DISCUSSION TITLE
Function to sort maps

### DISCUSSION THREAD

**Discussion by pepa65** [vlang/v#18649] [Ideas]
### Describe the feature

Maps have a unique pairing between key and value. It would be great if a map could be sorted in a way that the pairing is preserved. Right now the workaround to do this is inelegant and a bit lengthy.

Ideally, the sort would be on the key (as they are unique).

### Use Case

I was encountering the need when working on Advent of Code 2022 day 7.

### Other Information

You can write out a sort function manually and create a new sorted map. A builtin (or part of `maps`) method would be great.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.3 c16549b

### Environment details (OS name and version, etc.)

Linux Mint 21.1

**Comment by smartiniOnGitHub** [comment]
Hi, as alternative otherwise what do you think about the addition of a SortedMap (or similar) in the datatypes module inside vlib, and the ability to create such instance from the given map) ?

**Comment by pepa65** [comment]
I noticed a `SortedMap` entry in the documentation, but I couldn't find out how it works.

I think to just have a dedicated function for `.sort()` that sorts maps by the content of their keys would be great. It makes sense to have it, it can have the same name as the sort for arrays, and it's not hard to implement (I would think).

**Comment by pepa65** [comment]
To expand on my use case (and to offer a work-around):
```
mut mymap := map[string]int{}  // the map
// Code that fills the map with data...

// Sort the keys of the map
mut mykeys := mymap.keys()
mykeys.sort()

// Now the values in map can be iterated over in the sorted order
for k in mykeys {println(mymap[k])}
```

It would be nicer to have this instead:
```
mut mymap := map[string]int{}  // the map
// Code that fills the map with data...

// Sort the keys of the map
mymap.sort()

// Directly access the sorted values in the map
for k, _ in mymap {println(mymap[k])}
```
Then no extra array is necessary to keep around to access the sorted map.
